java/client/src/main/java/glide/api/commands/ClusterBaseCommands.java

```diff
@@ -0,0 +1,31 @@
+package glide.api.commands;
+
+import glide.api.models.configuration.Route;
+import java.util.concurrent.CompletableFuture;
+
+/**
+ * Base Commands interface to handle generic command and transaction requests with routing options.
+ *
+ * @param <T> The data return type.
+ */
+public interface ClusterBaseCommands<T> extends BaseCommands<T> {
```

Comment:
I'd need to see how this will work for StringCommands, HashCommands, etc. 
I think this model is overly complex just to save a single line in an interface.  And if we have several types (StringCommands will need around 4 generic types defined) this starts to look really ugly.

Reply:
Changed in 62cf8384